### PR TITLE
Initial tests for BigInt.prototype.toLocaleString().

### PR DESCRIPTION
**Pull request by Ms2ger**
Many of these tests are heavily based on the tests for
Number.prototype.toLocaleString().

### CODE REVIEW

**Comment by Ms2ger** [comment]
One bug in V8: it doesn't call `ThisBigIntValue()` in `toLocaleString()`.

CC @gsathya @FrankYFTang

File: test/intl402/BigInt/prototype/toLocaleString/builtin.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2012-2019 Mozilla Corporation; Igalia, S.L. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
```
Comment (FrankYFTang) [inline test/intl402/BigInt/prototype/toLocaleString/builtin.js]: I got the following error
"No license found in: intl402/BigInt/prototype/toLocaleString/builtin"

File: test/intl402/BigInt/prototype/toLocaleString/taint-Intl-NumberFormat.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2013-2019 Mozilla Corporation; Igalia, S.L. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
```
Comment (FrankYFTang) [inline test/intl402/BigInt/prototype/toLocaleString/taint-Intl-NumberFormat.js]: Got a "No license found in: intl402/BigInt/prototype/toLocaleString/taint-Intl-NumberFormat"
I think you need to change the line "// This code is governed by the license found in the LICENSE file." to
something like "// This code is governed by the BSD license found in the LICENSE file."

File: test/intl402/BigInt/prototype/toLocaleString/this-value-invalid.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  NaN,
+  -Infinity,
+  {valueOf: function () { return 5; }},
+  {valueOf: function () { return 5n; }},
+];
+
+for (const value of invalidValues) {
+  assert.throws(TypeError, function() {
+    BigInt.prototype.toLocaleString.call(value);
+  }, "BigInt.prototype.toLocaleString did not reject this = " + value + ".");
```
Comment (FrankYFTang) [inline test/intl402/BigInt/prototype/toLocaleString/this-value-invalid.js]: the ' + value + ' won't work for the test case of Symbol()

**Comment by FrankYFTang** [comment]
> One bug in V8: it doesn't call `ThisBigIntValue()` in `toLocaleString()`.
> 
> CC @gsathya @FrankYFTang

Just FYI
https://bugs.chromium.org/p/v8/issues/detail?id=9109 for v8 bug and
https://chromium-review.googlesource.com/c/v8/v8/+/1559214 for fix in v8

File: test/intl402/BigInt/prototype/toLocaleString/this-value-invalid.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-bigint.prototype.tolocalestring
+description: Tests that toLocaleString handles "thisBigIntValue" correctly.
+features: [BigInt]
+---*/
+
+var invalidValues = [
+//  undefined,
```
Comment (FrankYFTang) [inline test/intl402/BigInt/prototype/toLocaleString/this-value-invalid.js]: Please either remove the // or remove the whole line

**Comment by littledan** [comment]
It would be nice to additionally include tests that:
- Some locale other than en-US works, e.g., using the appropriate thousands separator
- Some value outside of Number range works, without rounding

**Comment by gsathya** [comment]
@Ms2ger I still get the missing license error that @FrankYFTang mentioned. These tests need to be updated:

```
No license found in: intl402/BigInt/prototype/toLocaleString/builtin
No license found in: intl402/BigInt/prototype/toLocaleString/default-options-object-prototype
No license found in: intl402/BigInt/prototype/toLocaleString/length
No license found in: intl402/BigInt/prototype/toLocaleString/returns-same-results-as-NumberFormat
No license found in: intl402/BigInt/prototype/toLocaleString/taint-Intl-NumberFormat
No license found in: intl402/BigInt/prototype/toLocaleString/this-value-invalid
No license found in: intl402/BigInt/prototype/toLocaleString/throws-same-exceptions-as-NumberFormat
```

**Comment by Ms2ger** [comment]
@gsathya I can't reproduce that locally on master. Could you clarify where you see those errors?

**Comment by Ms2ger** [comment]
More tests: https://github.com/tc39/test262/pull/2136

**Comment by gsathya** [comment]
> @gsathya I can't reproduce that locally on master. Could you clarify where you see those errors?

@Ms2ger, these tests break the license regexp in the test262 harness: 
https://github.com/test262-utils/test262-harness-py/blob/master/src/parseTestRecord.py#L27

**Comment by Ms2ger** [comment]
https://github.com/test262-utils/test262-harness-py/pull/6

**Pull request merged**